edge-hub/core/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/Cloud2DeviceTest.cs

```diff
@@ -204,6 +169,16 @@ ITransportSettings[] GetTransportSettings(TransportType transportType = Transpor
             return settings;
         }
 
+        async Task WaitForDeviceDisconnected(RegistryManager rm, string deviceName)
+        {
+            var device = await rm.GetDeviceAsync(deviceName);
+            while (device.ConnectionState != DeviceConnectionState.Disconnected)
+            {
+                device = await rm.GetDeviceAsync(deviceName);
+                await Task.Delay(TimeSpan.FromSeconds(5));
```

Comment:
Why do we need it here? Is there a way to do it without delay?

Reply:
I think there should be a delay so we don't make the calls to IotHub continuously. I picked 5s, it could be changed to lower value, maybe 3s?

Reply:
long term planning question: Could these tests accomplish their goals if we had a mock iothub and we didn't have to worry about network calls? We just make some situation or set of events that should occur in the mock iothub?

If so, then I think having some delays here in these tests are fine. If not, then I like the idea of trying to have as small a delay as possible. :)

Reply:
I think we need to ask IotHub on their SLA for operations. 5sec is too much IMO. If I could, I'd probably do it with interval not more that 1s, and total timeout 10s max.

Reply:
we can mock the sdk, so it won't connect to IotHub, but probably this issue wouldn't be caught. We should discuss in the tests improvements plan what should be mocked.